lab_3/main.py

```diff
@@ -59,49 +87,98 @@ def encode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
     :param corpus: a tuple of sentences
     :return: a tuple of the encoded sentences
     """
-    pass
+    if not isinstance(storage, LetterStorage) or not isinstance(corpus, tuple):
+        return ()
+
+    encoded_corpus = []
+    for sentence in corpus:
+        if isinstance(sentence[0], tuple):
+            sentence_list = []
+            for word in sentence:
+                sentence_list.append(tuple([storage.get_id_by_letter(letter) for letter in word]))
+            encoded_corpus.append(tuple(sentence_list))
+        else:
+            encoded_corpus.append(tuple([storage.get_id_by_letter(letter) for letter in sentence]))
+
+    if len(encoded_corpus) == 0:
```

Comment:
why there is such check for corpus length? we still return `tuple(encoded_corpus)` in the line 105